parachain/primitives/core/src/operating_mode.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use frame_support::{
+	dispatch::DispatchResult, traits::EnsureOrigin, PalletError, RuntimeDebug, StorageValue,
+};
+use scale_info::TypeInfo;
+use sp_std::fmt::Debug;
+
+/// Error generated by the `OwnedBridgeModule` trait.
+#[derive(Encode, Decode, TypeInfo, PalletError)]
+pub enum BridgeModuleError {
+	/// All pallet operations are halted.
+	Halted,
+}
+
+/// Operating mode for a bridge module.
+pub trait OperatingMode: Send + Copy + Debug + FullCodec {
+	// Returns true if the bridge module is halted.
+	fn is_halted(&self) -> bool;
+}
+
+/// Basic operating modes for a bridges module (Normal/Halted).
+#[derive(
+	Encode,
+	Decode,
+	Clone,
+	Copy,
+	PartialEq,
+	Eq,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+	serde::Serialize,
+	serde::Deserialize,
```

Comment:
I think this is a valid usage following the guide in https://serde.rs/no-std.html

Btw: binary built successful and tested the `setOperatingMode(operatingMode)` extrinsic work as expected.

Reply:
> I think this is a valid usage following the guide in https://serde.rs/no-std.html, also some other examples in polkadot-sdk

It is not valid usage based on the totality of evidence that I see. Serde is only needed for client code. Regardless of whether serde supports `no-std` or not, we do not want it in our `no-std` builds, because it increases the size of WASM binaries which are uploaded to the relay chain.

The evidence that you saw in the polkadot-sdk repo doesn't really mean anything. Perhaps the dev made a mistake and we should file a bug report.

In all our other other crates, serde is an optional dependency. For example we use `cfg_attr` below to optionally use serde:
https://github.com/Snowfork/snowbridge/blob/08ff3d63adbd2427fc71e2c7a6d364f09030628b/parachain/primitives/beacon/src/types.rs#L155

So I am not seeing a good reason why we should make an exception for this crate and not all the other crates.

Reply:
> So I am not seeing a good reason why we should make an exception for this crate and not all the other crates.

All make sense, fix in https://github.com/Snowfork/snowbridge/pull/969/commits/2e5ccf8793f21ccae28a56b47ad62fd6421579a4